cpu/sam0_common/periph/i2c.c

```diff
@@ -79,6 +79,20 @@ void i2c_init(i2c_t dev)
     int32_t tmp_baud;
 
     assert(dev < I2C_NUMOF);
+
+    /* i2c_deinit() has been called - restore config */
+    if (mutex_trylock(&locks[dev]) == 0) {
```

Comment:
Is it safe to call `mutex_trylock()` on uninitialized mutexes?

Reply:
As long as `MUTEX_INIT` remains [defined to `{{ NULL }}`](https://doc.riot-os.org/group__core__sync__mutex.html#ga96be0bfc33e7e113099c7546798bec99) and `locks` is a `static` buffer…

I agree it's not nice to rely on an implementation detail here, should probably make that more explicit (and drop the call to `mutex_init()` in `i2c_init()`.